### PR TITLE
Feature logarithmic weight adjustment

### PR DESCRIPTION
**Pull request by gilben1**
Cards this pull request addresses:
Vision -- Add Log Scaling to Penalty / Reward of Weights

To test:

Pull feature_logarithmic_weight_adjustment
Checkout master
Merge feature_logarithmic_weight_adjustment in
Catkin make
Test on robot

If you review this PR, please add your name and email below.

Reviewers:

### CODE REVIEW

File: op3_ball_detector/include/op3_ball_detector/ball_detector_config.h [hunk 1]
```diff
@@ -52,10 +53,13 @@ const int X_MIN_DEFAULT = 0;
 const int X_MAX_DEFAULT = 4000;
 const double LIGHT_SLOPE_DEFAULT = 1;
 const double LIGHT_CONSTANT_DEFAULT = 0;
-const int NUM_INTERVALS = 10; // test at 10
+const int NUM_INTERVALS = 20; // test at 10
 const double RANDOM_SAMPLE_CHANCE = 0.05;
-const int DETECTION_REWARD = 4;
-const int DETECTION_PENALTY = -1;
+const double DETECTION_REWARD_SCALAR = 4.0;
+const int DETECTION_PENALTY_SCALAR = -4.0;
```
Comment (mel-flygare96) [inline op3_ball_detector/include/op3_ball_detector/ball_detector_config.h]: DETECTION_PENALTY_SCALAR should also be a const double

File: op3_ball_detector/include/op3_ball_detector/ball_detector_config.h [hunk 2]
```diff
@@ -123,7 +127,7 @@ class BallColorConfig
   int sampleLightVal();
   int getMedianRVal(int x_val);
   void updateDistribution(std::vector<double> light_range, std::vector<double> light_weights);
-  void adjustWeightsWithLightVal(int light_val, int adjust_val, std::vector<double> &light_weights);
+  void adjustWeightsWithLightVal(int light_val, bool reward, std::vector<double> &light_weights);
```
Comment (mel-flygare96) [inline op3_ball_detector/include/op3_ball_detector/ball_detector_config.h]: I think something like bool is_reward is a bit more descriptive of the variable

**Review by mel-flygare96 — CHANGES_REQUESTED** [review]
Some change requests

File: op3_ball_detector/include/op3_ball_detector/ball_detector_config.h (hunk 1 above)
Comment (gilben1) [reply op3_ball_detector/include/op3_ball_detector/ball_detector_config.h]: Changed in ea0f08d

File: op3_ball_detector/include/op3_ball_detector/ball_detector_config.h (hunk 2 above)
Comment (gilben1) [reply op3_ball_detector/include/op3_ball_detector/ball_detector_config.h]: Changed in ea0f08d

**Review by mel-flygare96 — COMMENTED** [review]
gucchi

**Review by mel-flygare96 — APPROVED** [review]

**Review by mel-flygare96 — APPROVED** [review]
Gucchi

**Pull request merged**